src/scripts/main.js

```diff
@@ -1,3 +1,25 @@
 'use strict';
 
-// write your code here
+// head and foot
+const thead = document.querySelector('thead');
+const tfoot = document.querySelector('tfoot');
+
+[thead, tfoot].forEach((el) => {
+  const lastElement = el.querySelector('tr').lastElementChild;
```

Comment:
Ensure that each 'thead' and 'tfoot' element contains at least one 'tr' element. If not, 'el.querySelector('tr')' will return null, causing 'lastElement' to be null and potentially leading to errors when 'lastElement.insertAdjacentElement' is called.